### PR TITLE
5 Editar perfil

### PR DESCRIPTION
**Pull request by maciasj**
# Description

## Context

(Optional)

Current situation of the system, explanation why this PR is needed.

## Goal

Explanation about what we want to achieve with this PR.

## Support documents

- Taiga Ticket: [ticket](link)

- UI/UX design: [link](https://paack.co)

## Design considerations

(Optional)

Section dedicated to explain what design considerations we had while coding this PR.

# Testing

Section dedicated to explain how and what to test in this PR.

# Configuration

## New environment

(Optional)

Environment variables added/changed that needs to be added manually by the other developers

## New dependencies

(Optional)

New packages/libraries added / updated

# Attachments

(Optional)

- <details closed>
    
  </details>

Section dedicated to add any screenshots or other documents we find relevant to help understand or analyze this PR.

# Dependencies or blockers

(Optional: List of dependencies or blockers that make this PR not yet ready to be merged with the main branch)
## Subtasks

(List of subtask PRs needed for this feature)

- [ ] #XX
...

### CODE REVIEW

File: src/application/controllers/user-controller/user-controller.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import type { Request, Response } from "express";
-
-import type { IUser } from "~/domain/entities/user";
+//import { MongoNetworkTimeoutError } from "mongodb";
+import { IUser } from "~/domain/entities/user";
 import { UserRepository } from "~/domain/repositories/user-repository/user-repository";
+//import { isNull } from "util";
```
Comment (quimvi) [inline src/application/controllers/user-controller/user-controller.ts:5]: Esto se puede borrar

File: src/application/controllers/user-controller/user-controller.ts [hunk 2]
```diff
@@ -34,4 +35,40 @@ export class UserController {
       });
     }
   }
+
+   
+  public static async editParam(_req: Request, res: Response): Promise<void> {
```
Comment (quimvi) [inline src/application/controllers/user-controller/user-controller.ts:40]: Y si llamamos la funcion editUser ?

File: src/domain/repositories/user-repository/user-repository.ts [hunk 3]
```diff
@@ -9,4 +9,30 @@ export class UserRepository {
   public static async getAllUsers(): Promise<IUser[]> {
     return await UserModel.find();
   }
+
+  public static async editParam(nouParam: string, oldParam: string, tipusAtribut: string ): Promise<void> {
```
Comment (quimvi) [inline src/domain/repositories/user-repository/user-repository.ts:13]: Esto creo que sera updateUser realmente, porque el repositorio lo unique que hace es pillar un user y updatearlo

File: src/domain/repositories/user-repository/user-repository.ts [hunk 4]
```diff
@@ -9,4 +9,30 @@ export class UserRepository {
...
+         await UserModel.findOneAndUpdate({ username: oldParam }, { username: nouParam });
+         break;
+      case "email":
+         await UserModel.findOneAndUpdate({ email: oldParam }, { email: nouParam });
+
+    }
+    }
+
+  
+  public static async existParam(param: string, tipusAtribut: string): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (maciasj) [reply src/domain/repositories/user-repository/user-repository.ts:25]: eso lo veía más como para comprobar que el username nuevo que fuese a elegir no estuviese ya, pero creo que seria más fácil haciendo unique el valor en concreto

**Comment by quimvinals** [comment]
Le mande un audio a Marta sobre como hacer esto. porsi quieres preguntar que te diga

**Comment by quimvi** [comment]
Si poses tests estara perfect ja :D

**Review by quimvi — APPROVED** [review]

**Pull request merged**